tests/functional/object/mcg/test_noobaa_db_tls.py

```
@@ -0,0 +1,389 @@
+            logger.error(
+                f"CNPG cert secret '{name}' has type '{secret_type}', "
+                f"expected '{expected['type']}'"
+            )
+            valid = False
+
+        data = secret.get("data", {})
+        missing_keys = [key for key in expected["keys"] if key not in data]
+        if missing_keys:
+            logger.error(
+                f"CNPG cert secret '{name}' is missing key material {missing_keys}; "
+                f"keys present = {sorted(data)}"
```

Ruff (0.16.2)

[warning] Logging statement uses f-string

(G004)